sdk/Managed/test/Microsoft.WindowsAzure.MobileServices.Test/UnitTests/Table/Sync/MobileServiceSyncContext.Test.cs

```diff
@@ -246,6 +246,136 @@ public class MobileServiceSyncContextTests : TestBase
             await service.SyncContext.PushAsync();
         }
 
+        class LocalStoreWithDelay : MobileServiceLocalStoreMock
+        {
+            private int _lookupDelayInMilliseconds = 0;
+
+            public override async Task<JObject> LookupAsync(string tableName, string id)
+            {
+                if (0 != _lookupDelayInMilliseconds)
+                {
+                    // releases this thread and causes this lookup to take longer
+                    await Task.Delay(_lookupDelayInMilliseconds);
+                }
+
+                return await base.LookupAsync(tableName, id);
+            }
+
+            /// <summary>
+            /// Tells this mock to yield and to delay for the given number of milliseconds before proceeding with the LookupAsync().
+            /// </summary>
+            /// <param name="delayInMilliseconds">The delay in milliseconds.</param>
+            internal void SetLookupDelay(int delayInMilliseconds)
+            {
+                _lookupDelayInMilliseconds = delayInMilliseconds;
+            }
+        }
+
+        [AsyncTestMethod]
+        public async Task PushAsync_Succeeds_WithPendingOperations_AndOpQueueIsConsistent()
```

Comment:
A simpler test would be to block the deletion of operation from the store and attempt an operation on the same item. If the operation succeeds then we have an error, otherwise test passes. You can put a timeout on second operation (because it won't get access to lock as the delete of operation is blocked).